app/src/main/java/net/squanchy/home/HomeActivity.java

```diff
@@ -143,6 +153,40 @@ protected void onStart() {
         }
     }
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode == MY_PERMISSIONS_REQUEST_ACCESS_FINE_LOCATION) {
+            if (hasGrantedFineLocationAccess(grantResults)) {
+                proximityService.startRadar();
+            }
+        }
+    }
+
+    private void askProximityPermissionToStartRadar() {
+        if (hasLocationPermission()) {
+            proximityService.startRadar();
+        } else {
+            ActivityCompat.requestPermissions(
+                    this,
```

Comment:
I'd abstract this to a permission service so we keep the logic in the activity to a minimum (maybe next pr?)